File: tiny-event-sourcing-spring-app/src/main/kotlin/ru/quipy/shop/payment/events/PaymentSetDateEvent.java
package ru.quipy.shop.payment.events;

import lombok.Getter;
import ru.quipy.domain.Event;
import ru.quipy.shop.payment.Payment;
import ru.quipy.shop.payment.PaymentAggregate;

import java.util.Date;
import java.util.UUID;

public class PaymentSetDateEvent extends Event<PaymentAggregate> {
    @Getter
    private Date date;
    @Getter
    private UUID paymentId;

    public PaymentSetDateEvent(UUID paymentId, Date date) {
        super();
        this.paymentId = paymentId;
        this.date = date;
    }
}

File: tiny-event-sourcing-spring-app/src/main/kotlin/ru/quipy/shop/payment/events/PaymentSetDateEvent.kt
package ru.quipy.shop.payment.events

import lombok.Getter
import ru.quipy.core.annotations.DomainEvent
import ru.quipy.domain.Event
import ru.quipy.shop.payment.PaymentAggregate
import java.util.*

@DomainEvent(name = "PAYMENT_SET_DATE")
class PaymentSetDateEvent(
    @Getter
    val date: Date,
    @Getter
    val paymentId: UUID
) : Event<PaymentAggregate>(
    name = "PAYMENT_SET_DATE"
)